### PR TITLE
fixes #21275 - unique identifier needed for content-override name

### PR DESCRIPTION
**Pull request by cfouant**
Requires https://github.com/Katello/katello/pull/7081

@parthaa if you don't mind being a reviewer on this as well...

### CODE REVIEW

**Comment by akofink** [comment]
Actually, I just realized, the name option should be the activation key name for ID resolution. That's basically why this bug was reported (it seems to me) because the user was confused about the name option, which behaves differently than any other hammer command.

That being said, the API never sees this activation key name param because we resolve the AK ID from it and pass that to the API. We could fix this bug simply by (not changing the API) changing the options in this command to `--name` for the AK name and something like `--override-name` for the name API param. Just a suggestion for discussion, not a request.

**Comment by cfouant** [comment]
Since the API hasn't actually hit any Katello release, upstream or down, Partha and I discussed this and thought the better solution would be to fix the API.

**Comment by cfouant** [comment]
@akofink Since the API hasn't actually hit any Katello release, upstream or down, Partha and I discussed this and thought the better solution would be to fix the API.

File: lib/hammer_cli_katello/content_override.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def request_params
           override = { 'content_label' => option_content_label }
           override['value'] = option_value if option_value
           override['remove'] = true if option_remove?
-          override['name'] = option_name
+          override['override_name'] = option_override_name
```
Comment (parthaa) [inline lib/hammer_cli_katello/content_override.rb]: This change is unnecessary imo. Cant you just do
```
override['name'] = option_override_name
```
and get away with it. So you keep the ```--override-name``` option but not change the API

**Comment by parthaa** [comment]
@akofink @cfouant might want to address my comment https://github.com/Katello/hammer-cli-katello/pull/530#pullrequestreview-79270145 before merging.. My suggestion would not require any change in the API . Just change the name of the flag and  update the request_params method in hammer.

**Comment by akofink** [comment]
@parthaa I agree with you that that would be the ideal change. I don't see the `name` param as a problem considering it's nested under `content_overrides`. If I had to vote, I'd vote to not change the param, but I don't feel strongly about it.

**Comment by cfouant** [comment]
@parthaa @akofink changes made in agreement! Please re-review and let me know if you would like any additional changes, thanks!

**Comment by parthaa** [comment]
ACK

**Pull request merged**